observability/src/test/java/io/grpc/observability/interceptors/ConfigFilterHelperTest.java

```diff
@@ -0,0 +1,229 @@
+
+    Set<EventType> expectedlogEventTypeSet = ImmutableSet.copyOf(configEventTypes);
+    assertEquals(configFilterHelper.logEventTypeSet, expectedlogEventTypeSet);
+  }
+
+  @Test
+  public void checkMethodAlwaysLogged() {
+    List<LogFilter> sampleFilters = Stream.of(
+            new LogFilter("*", 4096, 4096))
+        .collect(Collectors.toList());
+    ImmutableList<LogFilter> sampleLogFilters =
+        ImmutableList.copyOf(sampleFilters);
+    configFilterHelper.setEmptyFilter();
+    when(mockConfig.getLogFilters()).thenReturn(sampleLogFilters);
+    configFilterHelper.setMethodOrServiceFilterMaps();
+
+    String fullMethodName = "service1/Method6";
+    MethodFilterParams expectedParams =
+        configFilterHelper.globalParams;
+    MethodFilterParams resultParams
+        = configFilterHelper.isMethodToBeLogged(fullMethodName);
+
+    assertEquals(resultParams.log, expectedParams.log);
+    assertEquals(resultParams.headerBytes, expectedParams.headerBytes);
+    assertEquals(resultParams.messageBytes, expectedParams.messageBytes);
+  }
+
+  @Test
+  public void checkMethodNotToBeLogged() {
+    List<LogFilter> sampleFilters = Stream.of(
+            new LogFilter("service1/Method2", 1024, 1024),
+            new LogFilter("service2/*", 2048, 1024))
+        .collect(Collectors.toList());
+    ImmutableList<LogFilter> sampleLogFilters =
+        ImmutableList.copyOf(sampleFilters);
+    configFilterHelper.setEmptyFilter();
+    when(mockConfig.getLogFilters()).thenReturn(sampleLogFilters);
+    configFilterHelper.setMethodOrServiceFilterMaps();
+
+    String fullMethodName = "service3/Method3";
+    MethodFilterParams expectedParams =
+        new MethodFilterParams(false, 0, 0);
+    MethodFilterParams resultParams
+        = configFilterHelper.isMethodToBeLogged(fullMethodName);
+
+    assertEquals(resultParams.log, expectedParams.log);
+    assertEquals(resultParams.headerBytes, expectedParams.headerBytes);
+    assertEquals(resultParams.messageBytes, expectedParams.messageBytes);
+  }
+
+  @Test
+  public void checkMethodToBeLoggedConditional() {
+    configFilterHelper.setEmptyFilter();
+    when(mockConfig.getLogFilters()).thenReturn(configLogFilters);
+    configFilterHelper.setMethodOrServiceFilterMaps();
+
+    String fullMethodName = "service1/Method2";
+    MethodFilterParams expectedParams =
+        new MethodFilterParams(true, 1024, 1024);
+    MethodFilterParams resultParams
+        = configFilterHelper.isMethodToBeLogged(fullMethodName);
+
+    assertEquals(resultParams.log, expectedParams.log);
+    assertEquals(resultParams.headerBytes, expectedParams.headerBytes);
+    assertEquals(resultParams.messageBytes, expectedParams.messageBytes);
+
+    String serviceNameWildcard = "service2/*";
+    MethodFilterParams expectedParamsWildCard =
+        new MethodFilterParams(true, 2048, 1024);
+    MethodFilterParams resultParamsWildCard
+        = configFilterHelper.isMethodToBeLogged(serviceNameWildcard);
```

Comment:
2 comments:

- you can just inline serverNameWildcard
- the pattern in the filter can contain wildcard but what does it mean to pass a wildcard like "service2/*" to the function `isMethodToBeLogged` ? This will always be an actual RPC method name and never a wildcard string, right?

Reply:
Updated `isMethodToBeLogged` to take `MethodDescriptor` as a parameter instead of String and updated fullMethodName to not use wildcard as done before.